### PR TITLE
Refactor Webhook

### LINKED ISSUE #888: [BUG] refactor webhook
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅱ. Does this pull request fix one issue?

fixes #888

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		client: c,
+		name:   NAME,
+	}
+}
+
+func (p *RequireNodeWithFuse) GetName() string {
+	return p.name
+}
+
+func (p *RequireNodeWithFuse) Mutate(pod *corev1.Pod, runtimeInfos []base.RuntimeInfoInterface) (shouldStop bool) {
```
Comment (allenhaozi) [inline pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go:48]: `shouldStop` is not assigned, only have an initial value
is this expected logic ?

Comment (cheyang) [reply pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go:48]: Yeah, the value is false by design.

File: pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	requiredSchedulingTerms = corev1.NodeSelectorTerm{
+		MatchExpressions: []corev1.NodeSelectorRequirement{},
+	}
+
+	if runtimeInfo == nil {
+		err = fmt.Errorf("RuntimeInfo is nil")
+		return
+	}
+
+	isGlobalMode, selectors := runtimeInfo.GetFuseDeployMode()
```
Comment (xieydd) [inline pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go:86]: why when fuse is not global, no selectors for fuse pod.

Comment (cheyang) [reply pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go:86]: When fuse is not global, the selector is the same with fluid worker.

File: pkg/common/label.go [hunk 3]
```diff
@@ -25,8 +25,8 @@ const (
 	// LabelAnnotationDatasetNum indicates the number of the dataset in specific node
 	LabelAnnotationDatasetNum = LabelAnnotationPrefix + "dataset-num"
 
-	// fluid adminssion webhook inject pod affinity strategy flag
-	LabelFluidSchedulingStrategyFlag = LabelAnnotationPrefix + "enable-scheduling-strategy"
+	// fluid adminssion webhook inject flag
+	EnableFluidInjectionFlag = LabelAnnotationPrefix + "enable-scheduling-strategy"
```
Comment (allenhaozi) [inline pkg/common/label.go:29]: Unified to `enable-injection` ? 
```go
	EnableFluidInjectionFlag = LabelAnnotationPrefix + "enable-injection"
```

Comment (cheyang) [reply pkg/common/label.go]: Yes. We'd like use feature gate for different functions. But we need a global flag to enable or disable webhook in general.

**Review by xieydd — APPROVED** [review]
/LGTM

File: docs/zh/operation/pod_schedule_default.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+正如亲和性所影响的，Pod会强制调度到了有缓存的node.172.16.0.16节点。
+
+```shell
+$ kubectl get pods nginx -o  custom-columns=NAME:metadata.name,NODE:.spec.nodeName
+NAME    NODE
+nginx   node.172.16.0.16
+```
+
+> 注释： K8s默认调度器并不需要配置强制亲和性，但是如果使用一些类似Volcano调度器组件并不感知K8s原生数据卷的亲和性，导致应用调度
```
[the remark this answers is not in the document]
Comment (cheyang) [reply docs/zh/operation/pod_schedule_default.md:175]: yeah. This PR is too large. I'd like to update docs in next PR.

**Pull request merged**